File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> table = new HashMap<>();
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (table.containsKey(diff)) {
                result[0] = i;
                result[1] = table.get(diff);
                return result;
            }
            table.put(nums[i], i);
        }
        return result;
    }
}

File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var table = HashMap<Int, Int>()
        nums.forEachIndexed { index, i ->
            val diff = target - i;
            if (table.containsKey(diff)) {
                return intArrayOf(index, table.get(diff)!!)
            }
            table.put(i, index)
        }
        return intArrayOf(0, 0)
    }
}